File: Lokomotif/lokomotif-kai/src/main/java/com/tujuhsembilan/lokomotifkai/model/Locomotive.java
package com.tujuhsembilan.lokomotifkai.model;

import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("locomotives")
public class Locomotive {
    @Id
    private String id;
    private String locoCode;
    private String locoName;
    private Double locoWeight;
    private Double locoLength;
    private Double locoWidth;
    private Double locoHeight;
    private String locoStatus;
    private String timestamps;
}

File: Lokomotif/lokomotif-be-express/schema/locomotive.js
const mongoose = require("mongoose");

const LocomotiveSchema = mongoose.Schema({
  locoCode: { type: String, required: true },
  locoName: { type: String, required: true },
  locoWeight: { type: Number, required: true },
  locoLength: { type: Number, required: true },
  locoWidth: { type: Number, required: true },
  locoHeight: { type: Number, required: true },
  locoStatus: { type: String, required: true },
  timestamps: { type: Date, required: true },
});

const Locomotive = mongoose.model("Locomotive", LocomotiveSchema);

module.exports = Locomotive;
